### PR TITLE
Add a database for caching timetable

### LINKED ISSUE #628: Database cache of timetable item
## Kind (Required)

- Improvement

 ## Overview (Required)

- We will put the timetable items in the Database and use it to display the data retrieved from the API.
- Basically, we will change the data from the Database data to flow through Coroutiens Flow and use it in the application.

https://github.com/DroidKaigi/conference-app-2021/blob/2bbcbed97b2c7a49540e0519ed2ccbbdcd235df3/data/repository/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/TimetableRepositoryImpl.kt#L10

 ## Links	

-

### PR DESCRIPTION
**Pull request by koji-1009**
## Issue
- close #628

## Overview (Required)
- Create TimetableItemDao
  - Add 3 queries
    - TimetableItemSession
    - TimetableItemSpecial
    - TimetableItemSpeaker
  - Add Select/Insert/Delete feature
- Improvement of TimetableRepositoryImpl

## Screenshot
- none

### CODE REVIEW

**Comment by takahirom** [comment]
👀

**Comment by takahirom** [comment]
Thank you. 👍 Implementation looks good 🐎 
When I tried app this PR, the app crashed. Because of DB migration.
I've made it possible to make destructive changes so you don't have to worry about migration.
Can you cherry-pick this commit? https://github.com/DroidKaigi/conference-app-2021/commit/edf1e068a97e549817a8ea8cf8de6dd1fd489b20 

```
git fetch https://github.com/DroidKaigi/conference-app-2021 edf1e068a97e549817a8ea8cf8de6dd1fd489b20
git cherry-pick edf1e068a97e549817a8ea8cf8de6dd1fd489b20
```

**Comment by takahirom** [comment]
Can you run `./gradlew :spotlessApply` ? 🙏 

```
Execution failed for task ':spotlessKotlinCheck'.
> The following files had format violations:
      data/db/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/TimetableItemDao.kt
          @@ -118,10 +118,12 @@
           ················),
           ················startsAt·=·Instant.fromEpochMilliseconds(row.startsAt),
           ················endsAt·=·Instant.fromEpochMilliseconds(row.endsAt),
          -················speakers·=·listOf(TimetableSpeaker(
          -····················name·=·row.speakerName,
          -····················iconUrl·=·row.speakerIconUrl,
          -················)),
          +················speakers·=·listOf(
          +····················TimetableSpeaker(
          +························name·=·row.speakerName,
          +························iconUrl·=·row.speakerIconUrl,
          +····················)
          +················),
           ············)
           ········}
           ········acc·+·mapOf(row.id·to·feedItem)
          @@ -147,10 +149,12 @@
           ················),
           ················startsAt·=·Instant.fromEpochMilliseconds(row.startsAt),
           ················endsAt·=·Instant.fromEpochMilliseconds(row.endsAt),
          -················speakers·=·listOf(TimetableSpeaker(
          -····················name·=·row.speakerName,
          -····················iconUrl·=·row.speakerIconUrl,
          -················)),
          +················speakers·=·listOf(
          +····················TimetableSpeaker(
          +························name·=·row.speakerName,
          +························iconUrl·=·row.speakerIconUrl,
          +····················)
          +················),
           ············)
           ········}
           ········acc·+·mapOf(row.id·to·feedItem)
      data/repository/src/main/java/io/github/droidkaigi/feeder/data/DataModule.kt
          @@ -59,8 +59,8 @@
           ····@Provides
           ····internal·fun·provideTimetableItemDao(timetableItemDao:·DaggerSQLDelightTimetableItemDao):
           ········TimetableItemDao·{
          -········return·timetableItemDao
          -····}
          +············return·timetableItemDao
          +········}
           
           ····@Provides
           ····internal·fun·provideTimetableRepository(daggerRepository:·DaggerTimetableRepositoryImpl):
  Run './gradlew :spotlessApply' to fix these violations.
```

**Comment by koji-1009** [comment]
@takahirom 
Thank you for your review!

File: data/repository/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/TimetableRepositoryImpl.kt [hunk 1]
```diff
@@ -1,27 +1,32 @@
...
     private val dataStore: UserDataStore,
 ) : TimetableRepository {
     override fun timetableContents(): Flow<TimetableContents> {
-        return flow { emit(droidKaigi2021Api.fetch()) }
-//        return dataStore.favorites()
-//            .combine(timetableItemDao.selectAll()) { favorites, dbtimetable ->
-//                timetableContents(dbtimetable.sortedByDescending { it.publishedAt }, favorites)
-//            }
+        return dataStore.favorites()
+            .combine(timetableItemDao.selectAll()) { favorites, dbtimetable ->
```
Comment (takahirom) [inline data/repository/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/TimetableRepositoryImpl.kt:16]: 🆒

**Review by takahirom — APPROVED** [review]
LGTM 💯

**Pull request merged**